### ISSUE TITLE
Display of `<expan>`

### ISSUE THREAD

**Issue by michaelnmmeyer** [erc-dharma/project-documentation#260]
I am not sure how I should display abbreviations, depending on the display mode. For:

```
 <expan><abbr>aug<am>g</am></abbr><ex>usti duo</ex></expan>
```

... I propose:

> Physical:
> Text: "augg"
> Tooltip: "Abbreviation (expansion: augusti duo)"
> 
> Logical:
> Text: "augusti duo"
> Tooltip: "Expanded text (abbreviation: augg)"
> 
> Full:
> Text: "auggusti duo"
> Tooltip:  "Abbreviation"

Is this ok?

We decided to use a background color for `<abbr>`. Should `<am>` have a peculiar format, too?

So far, `<am>` is not used, and we only find the pattern `<expan><abbr>...</abbr><ex>...</ex></expan>`, with a single occurrence of `<abbr>` + `<ex>`.

**Comment by arlogriffiths** [comment]
I never use expansions so I am not directly concerned... I approve of what you propose for Physical and Logical, but I don't see why display should be so undiplomatic in Full. I'd rather the abbreviation be displayed as such with some bracket or symbol marking that it is an abbreviation. The tooltip is not necessary in this display which is intended mainly for copy/pasting to print context.

**Comment by danbalogh** [comment]
I have never used expansions either, and especially the form with an "abbreviation marker" that is not part of the expanded abbreviation is expected to be very rare to nonexistent. It was included in the EGD just in case it occurs one day, because the provision exists in EpiDoc.

`<abbr>` plus `<ex>` without `<expan>` is an encoding error; `<expan>` must be present if the abbreviation is expanded.

Now some general thoughts. I think it is very important that expansions, if they are displayed at all, should be clearly indicated as non-original, preferably with hard markup (brackets of some sort), and not just colour. This applies to logical display as well as to the full one.
I also think that even in logical view, things present in the original (as the `<am>` in this example) should never be suppressed.
I do not think it is a good idea to show the contents of `<abbr>` in any kind of brackets, which would in my opinion be confusing to the reader. Brackets should be reserved for philological markup, and this is semantic business. If Arlo (and/or others) consider it important to have some sort of hard markup for abbreviations in any display, then a dagger or other symbol might be used, perhaps before rather than after the abbreviation, similarly to how personal and place names are indicated in the display.

Given the above, I propose the following, in all display modes:
- display contents of `<abbr>` (and `<am>` if any) with light brown (or whatever we agreed on) background colour. The contents of these elements should always be displayed in every mode.
- for `<abbr>` on its own (without `<expan>` and therefore without any other related elements), show tooltip with text "`<abbr>` is an abbreviation." E.g. "aug is an abbreviation."
- for `<abbr>` within `<expan>`, with mandatorily present `<ex>` and optionally present `<am>`, compose the tooltip like this: "`<abbr><am>` is an abbreviation of `<abbr><ex>`." E.g. "aug is an abbreviation of augusti" (for `<expan><abbr>aug</abbr><ex>usti</ex></expan>`) or "augg is an abbreviation of augusti duo" (for `<expan><abbr>aug</abbr><am>g</am><ex>usti duo</ex></expan>`)
- note that I don't think the tooltip should be eliminated in mixed mode. Its presence does no harm.

Beyond this, I'm not sure how far we should go. In my opinion, nothing beyond the above should be done in diplomatic view. I could imagine showing a dagger, as I suggest above, next to the abbreviation in logical and mixed mode,
I'm not at all sure that the contents of `<ex>` should ever be displayed in any mode other than through a tooltip. I would slightly prefer never to display it, but if Arlo or anyone else thinks displaying the expansion is essential, then I'm OK with showing it in logical and mixed mode. If it is shown, I propose that it should appear without the background colour for abbreviations, but with grey font colour, in angle brackets ⟨⟩, and in subscript. This would make the display similar to that of line numbers and milestones as well as to that of editorial emendations (which in my opinion makes intuitive sense), but the subscript would distinguish this from both of those things.
If this kind of display for `<ex>` is approved, then I further propose that simultaneously `<am>`, if it ever occurs, should be shown in superscript (without any brackets or suchlike).

**Comment by danbalogh** [comment]
It seems that we now have display for abbr and expan. See e.g. line 1r8=8 in https://dharmalekha.info/texts/INSTiruvavatuturai00047
I can't check the display of `<am>` because it doesn't occur in any of our editions and I don't know how I can see a display of https://github.com/erc-dharma/test/blob/03fe53e5ed5c296d3404ee5d770492d53e24f189/DHARMA_INSTestEditorial.xml which contains a test for it.
Has this thread been resolved satisfactorily?
I think yes and it should be closed.

**Comment by arlogriffiths** [comment]
the use of (...) for the expanded part seems inconsistent with https://dharmalekha.info/editorial-conventions. would use of ⟨...⟩ not be more consistent with our general conventions?

**Comment by danbalogh** [comment]
Indeed, my preference is definitely for ⟨...⟩. I don't want to worry overmuch about cosmetic details, but if the change is not too much trouble, it should be made.
We need input from @michaelnmmeyer : do we have display for `<am>`? is there a way for me to check the display of the test file?

**Comment by michaelnmmeyer** [comment]
Nothing special is done with `<am>`. The contents of the element is displayed as is, without any formatting.

So far, nobody used it. I however added test cases in [this test file](https://github.com/erc-dharma/test/blob/master/DHARMA_INSTestEditorial.xml) (under the section "abbr").

**Comment by danbalogh** [comment]
You have not answered my second question: is there a way for me to check the display of the test file? I'd like to see how the text and the associated tooltip looks in physical and logical view.

**Comment by michaelnmmeyer** [comment]
You need to use the [conversion script](https://github.com/erc-dharma/project-documentation/blob/master/stylesheets/convert.py) I wrote for this purpose. To invoke it, run `python3 /path/to/convert.py /path/to/the/xml_file.xml`. This performs the transformation immediately and opens the resulting Web page in your Web browser.

You can also set things up in Oxygen by following [these guidelines](https://github.com/erc-dharma/project-documentation/blob/master/stylesheets/README.md) (under the heading "How to display your inscription as on dharmalekha.info").

**Comment by danbalogh** [comment]
Thanks. I don't have write access to the test folder, so I could only make changes offline. For future reference, I suggest adding the following examples after augusti duo:
```
	<expan><abbr>mā</abbr><ex>ṣa</ex></expan>
	<expan><ex>jā</ex><abbr>k·</abbr> </expan> tandoṅ
	<expan><abbr>I</abbr><ex>raṇṭu</ex><abbr>m</abbr></expan>
```

I find the tooltips good.

For the actual display rendition, I would prefer the following changes:

- as per your original proposal at the top of this thread,
  - In Physical display, only the content of `<abbr>` and `<am>` should be shown; `<ex>` should be hidden.
  - in Logical display, only the content of `<abbr>` and `<ex>` should be shown.; `<am>` should be hidden.
  - in Full display, all three of these elements should be shown.
- additional details
  - when `<expan>` is present, the associated tooltip should be the same in all three views
  - the contents of `<ex>` should be wrapped in angle brackets ⟨...⟩ instead of parentheses () and also be shown in blue, just like `<reg>`

**Comment by michaelnmmeyer** [comment]
Implemented in https://github.com/erc-dharma/dharma/commit/d34341e486063603dd7333b33c96c1548fef8090.

**Comment by danbalogh** [comment]
Thanks. I believe this can be closed now.
